### PR TITLE
Setup tilt for etcd snapshot restore controller

### PR DESCRIPTION
**Pull request by alexander-demicev**
**What this PR does / why we need it**:

This PR adds support for Tilt for etcd backup restore controller and also moves all related kustomize config to a separate directory.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Related to: #503

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -242,9 +242,9 @@ generate-exp-etcdrestore-manifests-api: controller-gen ## Generate ClusterRole a
 	$(CONTROLLER_GEN) rbac:roleName=manager-role crd paths="./exp/etcdrestore/api/..." \
 			paths=./exp/etcdrestore/controllers/... \
 			paths=./exp/etcdrestore/webhooks/... \
-			output:crd:artifacts:config=./config/exp/etcdrestore/crd/bases \
-			output:rbac:dir=./config/exp/etcdrestore/rbac \
-			output:webhook:dir=./config/exp/etcdrestore/webhook \
+			output:crd:artifacts:config=./exp/etcdrestore/config/crd/bases \
+			output:rbac:dir=./exp/etcdrestore/config/rbac \
+			output:webhook:dir=./exp/etcdrestore/config/webhook \
```
Comment (Danil-Grigorev) [inline Makefile:247]: There is also CRD generation on line 526 which uses config/exp path, which needs to be changed https://github.com/rancher/turtles/pull/689/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R526

Comment (furkatgofurov7) [reply Makefile:247]: Yes, exactly. Also please run `make generate`

File: exp/etcdrestore/config/default/kustomization.yaml [hunk 2]
```diff
@@ -1,20 +1,14 @@
 # Adds namespace to all resources.
 namespace: rancher-turtles-system
 
-namePrefix: rancher-turtles-
+namePrefix: rancher-turtles-etcdbackuprestore
```
Comment (furkatgofurov7) [inline exp/etcdrestore/config/default/kustomization.yaml:4]:
```suggestion
namePrefix: rancher-turtles-etcdsnapshotrestore
```

Comment (furkatgofurov7) [reply exp/etcdrestore/config/default/kustomization.yaml:4]: Nit: We need to use the same CRD name as in the proposal throughout the code (Tiltfile, manager_image_patch.yaml etc), but that can be a follow-up PR.

**Review by furkatgofurov7 — APPROVED** [review]
Thanks!

**Comment by furkatgofurov7** [comment]
@alexander-demicev this partially fixes/related https://github.com/rancher/turtles/issues/503 so I updated PR description.

File: charts/rancher-turtles/templates/rancher-turtles-exp-etcdrestore-components.yaml [hunk 3]
```diff
@@ -103,21 +99,404 @@ spec:
...
+              fieldPath: metadata.namespace
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_UID
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.uid
+        image: ghcr.io/rancher/turtles-etcd-backup-restore:dev
```
Comment (furkatgofurov7) [inline charts/rancher-turtles/templates/rancher-turtles-exp-etcdrestore-components.yaml:411]:
```suggestion
        image: ghcr.io/rancher/turtles-etcd-snapshot-restore:dev
```

File: exp/etcdrestore/config/default/manager_image_patch.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: controller-manager
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+        - image: ghcr.io/rancher/turtles-etcd-backup-restore:dev
```
Comment (furkatgofurov7) [inline exp/etcdrestore/config/default/manager_image_patch.yaml:10]:
```suggestion
        - image: ghcr.io/rancher/turtles-etcd-snapshot-restore:dev
```

File: Tiltfile [hunk 5]
```diff
@@ -50,6 +50,19 @@ projects = {
         ],
         "kustomize_dir": "config/default",
         "label": "turtles"
+    },
+    "turtles-etcdsnapshotrestore": {
+        "context": "exp/etcdrestore",
+        "image": "ghcr.io/rancher/turtles-etcd-backup-restore:dev",
```
Comment (furkatgofurov7) [inline Tiltfile:56]:
```suggestion
        "image": "ghcr.io/rancher/turtles-etcd-snapshot-restore:dev",
```

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

**Review by furkatgofurov7 — APPROVED** [review]

**Review by Danil-Grigorev — APPROVED** [review]

**Pull request merged**